cabal.project

```diff
@@ -13,6 +13,8 @@ repository cardano-haskell-packages
     c00aae8461a256275598500ea0e187588c35a5d5d7454fb57eac18d9edb86a56
     d4a35cd3121aa00d18544bb0ac01c3e1691d618f462c46129271bccf39f7e8ee
 
+-- repeat hackage index-state to work around haskell.nix parsing limitation
+index-state: 2022-10-24T00:00:00Z
```

Comment:
Is it important to repeat it? Should we not just have two `index-state` stanzas, one for hackage and one for CHaP?

Reply:
Long story short is that if you do
```
index-state: 2022-10-24T00:00:00Z
index-state: cardano-haskell-packages 2022-11-01T00:00:00Z
```
you end up with hackage at HEAD since hackage is not mentioned in the second standa and HEAD is the default ("the second index-state stanza completely ovverides the first"). The correct syntax to set both index-states is 
```
index-state:
  -- Bump this if you need newer packages from Hackage
  , hackage.haskell.org 2022-10-24T00:00:00Z
  -- Bump this if you need newer packages from CHaP
  , cardano-haskell-packages 2022-11-01T00:00:00Z
```

BUT haskell.nix currently cannot parse this syntax so the best solution is to do

```
-- this is for haskell.nix (haskell.nix will only read the first one and this is the only syntax it accepts)
index-state: 2022-10-24T00:00:00Z
-- this is for cabal (cabal will ignore the first and use the second anyway)
index-state:
  -- Bump this if you need newer packages from Hackage
  , hackage.haskell.org 2022-10-24T00:00:00Z
  -- Bump this if you need newer packages from CHaP
  , cardano-haskell-packages 2022-11-01T00:00:00Z
```

Reply:
It is a mess? yes. Is everything broken? also yes.